src/auction.js

```diff
@@ -0,0 +1,518 @@
+
+      const sizes = adUnit && adUnit.bids && adUnit.bids[0] && adUnit.bids[0].sizes;
+      const parsedSizes = utils.parseSizesInput(sizes);
+
+      // if a banner impression has one valid size, we assign that size to any bid
+      // response that does not explicitly set width or height
+      if (parsedSizes.length === 1) {
+        const [ width, height ] = parsedSizes[0].split('x');
+        bid.width = width;
+        bid.height = height;
+        return true;
+      }
+
+      return false;
+    }
+
+    // Postprocess the bids so that all the universal properties exist, no matter which bidder they came from.
+    // This should be called before addBidToAuction().
+    function prepareBidForAuction() {
+      let bidRequest = _bidderRequests.find(request => {
+        return request.bids
+          .filter(rbid => rbid.bidder === bid.bidderCode && rbid.adUnitCode === adUnitCode).length > 0;
+      }) || {start: null};
+
+      const start = bidRequest.start;
+
+      Object.assign(bid, {
+        auctionId: _auctionId,
+        requestId: bidRequest.requestId,
+        responseTimestamp: timestamp(),
+        requestTimestamp: start,
+        cpm: parseFloat(bid.cpm) || 0,
+        bidder: bid.bidderCode,
+        adUnitCode
+      });
+
+      bid.timeToRespond = bid.responseTimestamp - bid.requestTimestamp;
+
+      // Let listeners know that now is the time to adjust the bid, if they want to.
+      //
+      // CAREFUL: Publishers rely on certain bid properties to be available (like cpm),
+      // but others to not be set yet (like priceStrings). See #1372 and #1389.
+      events.emit(CONSTANTS.EVENTS.BID_ADJUSTMENT, bid);
+
+      // a publisher-defined renderer can be used to render bids
+      const adUnitRenderer =
+        bidRequest.bids && bidRequest.bids[0] && bidRequest.bids[0].renderer;
+
+      if (adUnitRenderer) {
+        bid.renderer = Renderer.install({ url: adUnitRenderer.url });
+        bid.renderer.setRender(adUnitRenderer.render);
+      }
+
+      const priceStringsObj = getPriceBucketString(
+        bid.cpm,
+        config.getConfig('customPriceBucket'),
+        config.getConfig('currency.granularityMultiplier')
+      );
+      bid.pbLg = priceStringsObj.low;
+      bid.pbMg = priceStringsObj.med;
+      bid.pbHg = priceStringsObj.high;
+      bid.pbAg = priceStringsObj.auto;
+      bid.pbDg = priceStringsObj.dense;
+      bid.pbCg = priceStringsObj.custom;
+
+      // if there is any key value pairs to map do here
+      var keyValues;
+      if (bid.bidderCode && (bid.cpm > 0 || bid.dealId)) {
+        keyValues = getKeyValueTargetingPairs(bid.bidderCode, bid);
+      }
+
+      // use any targeting provided as defaults, otherwise just set from getKeyValueTargetingPairs
+      bid.adserverTargeting = Object.assign(bid.adserverTargeting || {}, keyValues);
+    }
+
+    function doCallbacksIfNeeded() {
+      if (bid.timeToRespond > $$PREBID_GLOBAL$$.cbTimeout + $$PREBID_GLOBAL$$.timeoutBuffer) {
```

Comment:
Not specifically for this PR, but... should these be on `config` @matthewlane?

If they make sense as global configs, then they probably should. If not, then you should accept them as params here. I don't think it makes sense to read from $$PREBID_GLOBAL$$ in either case, though.

If they are moving to `config`, that change should go in `master`.